### DISCUSSION TITLE
Dynamic page titles?

### DISCUSSION THREAD

**Discussion by tjakobi** [zauberzeug/nicegui#830] [Q&A] [answered]
Dear nicegui developers,

I was wondering if it is possible to modify a page title while generating the page, specifically based on input from the path, something like:

```py
@ui.page('/detail/{item_id}', title="static title")
    async def page_application_display_detail(item_id: str):

    # do something

    page_title = "new dynamic title" + item_id

```
The idea would be to have the page respond to user input such as "Your search for item XZY" instead of having to set a fixed title.

Maybe this is already possible, but I could not find any related code/discussion/issues.

Thank you!

**Comment by rodja** [comment, marked as answer]
Sure. You can run javascript as soon as the client has connected:

```py
from nicegui import ui, Client

@ui.page('/')
@ui.page('/{path:path}')
async def index(path: str, client: Client):
    await client.connected()
    await ui.run_javascript(f'document.title = "{path or "nothing"}";', respond=False)

ui.run()
```

**Reply by tjakobi** [reply]
Update April 2024, @rodja's code still works, but the `respond=False` argument has to now be removed:

`ValueError: The "respond" argument of run_javascript() has been removed. Now the function always returns an AwaitableResponse that can be awaited. Please remove the "respond=False" argument and call the function without awaiting.`

Not sure which version of nicegui triggered this change though.

**Reply by falkoschindler** [reply]
Thanks, @tjakobi! I think it was version 1.4.0, see PR https://github.com/zauberzeug/nicegui/pull/1797.

**Comment by tjakobi** [comment]
Thank you very much, works beautifully!

**Comment by tjakobi** [comment]
Potentially helpful for users in the future regarding this topic: 

While the <title> is now shown correctly in the browser window, my main reason for having specific title tags is to improve visibility in search engines.  I was wondering if this change (which happens on the client site) would be reflected in search hits generated by Google and Bing.

As it turns out, search engines may execute the client side javascript to improve results, Google seems to be among them: https://stackoverflow.com/questions/30588501/seo-affected-changing-title-tag-by-javascript

As such, this approach should work for the sake of SEO. I will report in some time, once my page has been picked up the the search engines.
